src/main/resources/META-INF/resources/primefaces-extensions/fuzzysearch/1-fuzzysearch-widget.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * PrimeFaces Fuzzy Search Widget
+ */
+PrimeFaces.widget.FuzzySearch = PrimeFaces.widget.BaseWidget.extend({
+
+    /**
+     * Initializes the widget.
+     * 
+     * @param {object}
+     *        cfg The widget configuration.
+     */
+    init: function (cfg) {
+        this._super(cfg);
+
+        this.input = $(this.jqId + '_fuzzysearch-search-input');
+        this.results = $(this.jqId + '_fuzzysearch-search-results');
+        this.keys = JSON.parse(cfg.keys);
+        this.datasource = JSON.parse(cfg.value);
+        this.resultStyle = cfg.resultStyle;
+        this.resultStyleClass = cfg.resultStyleClass;
+        this.items = this.results.children();
+
+        //Visual effects
+        PrimeFaces.skinInput(this.input);
+
+        this.bindKeyEvents();
+        this.bindDynamicEvents();
+    },
+
+    bindKeyEvents: function () {
+        var $this = this;
+
+        this.input.on('input', function (e) {
+            $this.search($this.input.val());
+        });
+    },
+
+    bindDynamicEvents: function () {
+        var $this = this;
+
+        this.items.on('click', function (e) {
+            if ($this.hasBehavior('select')) {
+                var ext = {
+                    params: [
+                        {name: this.id + '_itemSelect', value: this.innerHTML}
```

Comment:
I think `this.id` needs to be `$this.id` does it?

Reply:
Corrected to $this.id, but it does not trigger onSelect in controller.

Reply:
$this gives component id. Shouldn't we use the selected item id?

Reply:
Yeah you are right the problem is you are missing this from your Java Renderer.

```java
   @Override
    public void decode(final FacesContext context, final UIComponent component) {
        decodeBehaviors(context, component);
    }
```

Reply:
now getting "java.lang.String cannot be cast to org.primefaces.extensions.showcase.model.fuzzysearch.FuzzySearchItem;"
Is that the reason why you offer `f:selectItems` to turn the results into an object?

Reply:
yep that is exactly why.  The more I looked at how you are doing it this is really  Select type component.